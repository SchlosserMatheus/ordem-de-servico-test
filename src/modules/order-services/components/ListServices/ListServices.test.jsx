import React from 'react'
import { shallow, mount } from 'enzyme'
import ListServices from './ListServices'
import sinon from 'sinon'

describe('Testing ListServices Component', () => {
 
  it('should show a message that any service was created', () => {
    const wrapper = shallow(
      <ListServices services={[]} 
        onClickServiceDeleteButton={()=> null}
        onClickServiceEditButton={()=> null}
      /> 
    ).dive();
    expect(wrapper.find('.no-services').length).toEqual(1);  
  });


  it('should test if there are two buttons with the property onClick', () => {
    const serviceProps = [
      {description: 'x', quantity: 1, value: 10},
      {description: 'y', quantity: 5, value: 120}
    ];
    
    const wrapper = shallow(
      <ListServices 
        services={serviceProps} 
        onClickServiceDeleteButton={()=> null}
        onClickServiceEditButton={()=> null}
      /> 
    ).dive();
    expect(wrapper.find('.delete-button[onClick]').length).toEqual(2)

  });


  it('should test if there are two buttons with the proprierty onClick', () => {
    const serviceProps = [
      {description: 'x', quantity: 1, value: 10},
      {description: 'y', quantity: 5, value: 120}
    ];
    
    const wrapper = shallow(
      <ListServices 
        services={serviceProps} 
        onClickServiceDeleteButton={()=> null}
        onClickServiceEditButton={()=> null}
      /> 
    ).dive();
    expect(wrapper.find('.edit-button[onClick]').length).toEqual(2)
  });
  
  
  it('should check snapshot version', () => {
    const serviceProps = [
      {description: 'x', quantity: 1, value: 10},
      {description: 'y', quantity: 5, value: 120}
    ];
    
    const wrapper = shallow(
      <ListServices 
        services={serviceProps} 
        onClickServiceDeleteButton={()=> null}
        onClickServiceEditButton={()=> null}
      /> 
    );
    expect(wrapper).toMatchSnapshot();
  });


  it('should test the event onClickServiceEditButton when the function handleClickDeleteButton is called', () => {
    const serviceProps = [
      {description: 'x', quantity: 1, value: 10},
      {description: 'y', quantity: 5, value: 120}
    ];
    
    const callbackClickDeleteButton = sinon.spy();
    const wrapper = shallow(
      <ListServices 
        services={serviceProps} 
        onClickServiceDeleteButton={callbackClickDeleteButton}
        onClickServiceEditButton={()=> null}
      /> 
    ).dive();
    
    const wrapperInstance = wrapper.instance();
    const mockInstanceConfirmAlert = sinon.stub(global, 'confirm');
    mockInstanceConfirmAlert.returns(true);

    wrapperInstance.handleClickDeleteButton(0);
    expect(callbackClickDeleteButton.callCount).toEqual(1);
    
    mockInstanceConfirmAlert.restore();

  });


  /**
  |--------------------------------------------------
  |     Tests using Mount to test events
  |--------------------------------------------------
  */

  it("should simulates delete event when clicked in the alert confirm", () => {
    const serviceProps = [
      {description: 'x', quantity: 1, value: 10},
      {description: 'y', quantity: 5, value: 120}
    ];
    
    const callbackClickDeleteButton = sinon.spy();

    const wrapper = mount(
      <ListServices 
        services={serviceProps} 
        onClickServiceDeleteButton={callbackClickDeleteButton}
        onClickServiceEditButton={()=> null}
      /> 
    );

    const deleteButton = wrapper.find('IconButton.delete-button').first();
    const confirmButtonDeleteService = sinon.stub(global, 'confirm');
    confirmButtonDeleteService.returns(true);
    deleteButton.simulate("click");
    expect(confirmButtonDeleteService.calledOnce).toEqual(true);

    confirmButtonDeleteService.restore();
  });
  

  it('should test the event onClick of button Edit', () => {
    const serviceProps = [
      {description: 'x', quantity: 1, value: 10},
      {description: 'y', quantity: 5, value: 120}
    ];
    
    const callbackClickEditButton = sinon.spy();
    const wrapper = mount(
      <ListServices 
        services={serviceProps} 
        onClickServiceDeleteButton={()=> null} 
        onClickServiceEditButton={callbackClickEditButton}
      /> 
    );
    wrapper.find('Button.edit-button').first().simulate('click');
    expect(callbackClickEditButton.callCount).toEqual(1)

    })
  
});

