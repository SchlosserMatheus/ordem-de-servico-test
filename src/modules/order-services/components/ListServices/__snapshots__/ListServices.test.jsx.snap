// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ListServices Component should check snapshot version 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(ListServices)
    onClickServiceDeleteButton={[Function]}
    onClickServiceEditButton={[Function]}
    services={
      Array [
        Object {
          "description": "x",
          "quantity": 1,
          "value": 10,
        },
        Object {
          "description": "y",
          "quantity": 5,
          "value": 120,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "root": "ListServices-root-1",
        "table": "ListServices-table-2",
      },
      "onClickServiceDeleteButton": [Function],
      "onClickServiceEditButton": [Function],
      "services": Array [
        Object {
          "description": "x",
          "quantity": 1,
          "value": 10,
        },
        Object {
          "description": "y",
          "quantity": 5,
          "value": 120,
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "root": "ListServices-root-1",
          "table": "ListServices-table-2",
        },
        "onClickServiceDeleteButton": [Function],
        "onClickServiceEditButton": [Function],
        "services": Array [
          Object {
            "description": "x",
            "quantity": 1,
            "value": 10,
          },
          Object {
            "description": "y",
            "quantity": 5,
            "value": 120,
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
